xarray/core/dataset.py

```diff
@@ -309,7 +309,7 @@ def _maybe_chunk(
             # when rechunking by different amounts, make sure dask names change
             # by providing chunks as an input to tokenize.
             # subtle bugs result otherwise. see GH3350
-            token2 = tokenize(name, token if token else var._data, chunks)
```

Comment:
It's the `chunks`, but we include the `name` in in the generated token just below anyway

Reply:
I'm OK merging but seems like it should be fixed in dask.

PS: I love that you're looking in to this! https://github.com/dask/dask/pull/10579

Reply:
> it should be fixed in dask.

I don't think that will happen. We are in a position to know that the str of the chunks tuple is unique, but you cannot know this for general objects where str() might not produce anything unique.

Reply:
Ah right. Can you add a comment to that effect? It'd be nice to add a benchmark too (separate PR is nice because then we can see the change here) but I can get to that later if you don't have the time now